src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -3033,11 +3017,10 @@ static void benign_reclaimer_locked(void* arg, grpc_error_handle error) {
 
 static void destructive_reclaimer_locked(void* arg, grpc_error_handle error) {
   grpc_chttp2_transport* t = static_cast<grpc_chttp2_transport*>(arg);
-  size_t n = grpc_chttp2_stream_map_size(&t->stream_map);
   t->destructive_reclaimer_registered = false;
-  if (error.ok() && n > 0) {
-    grpc_chttp2_stream* s = static_cast<grpc_chttp2_stream*>(
-        grpc_chttp2_stream_map_rand(&t->stream_map));
+  if (error.ok() && !t->stream_map.empty()) {
+    // As stream_map is a hash map, this selects effectively a random stream.
+    grpc_chttp2_stream* s = t->stream_map.begin()->second;
```

Comment:
I am not sure this is random, but I'm not sure it matters

Reply:
Random enough for the intended purpose!